test/compile/legend/encode.test.ts

```diff
@@ -32,6 +32,27 @@ describe('compile/legend', function() {
         }), COLOR, {} as LegendComponent);
         assert.deepEqual(symbol.shape.value, 'square');
     });
+
+    it('should have default opacity', function() {
+      const symbol = encode.symbols({field: 'a', type: 'nominal'}, {}, parseUnitModelWithScale({
+          mark: "point",
+          encoding: {
+            x: {field: "a", type: "nominal"}}
+        }), COLOR, {} as LegendComponent);
+        assert.deepEqual(symbol.opacity.value, 0.7); // default opacity is 0.7.
+    });
+  });
+
+  describe('encode.gradient', function() {
+    it('should have default opacity', function() {
+      const gradient = encode.gradient({field: 'a', type: 'quantitative'}, {}, parseUnitModelWithScale({
+          mark: "point",
+          encoding: {
+            x: {field: "a", type: "quantitative"}}
+        }), COLOR, {"implicit":{"type":"gradient"}} as LegendComponent);
```

Comment:
Please don't do unnecessary  `as` here.   You should instead properly create an instance of LegendComponent.